File: 2563. Count the Number of Fair Pairs/Solution.py
class Solution:
    def countFairPairs(self, nums: List[int], lower: int, upper: int) -> int:
        nums = sorted(nums)
        return self.countLess(nums, upper + 1) - self.countLess(nums, lower)
    
    def countLess(self, nums, val):
        res, left, right = 0, 0, len(nums) - 1
        while left < right:
            while left < right and nums[left] + nums[right] >= val: right -= 1
            res += right - left
            left += 1
        return res

File: 2563. Count the Number of Fair Pairs/Solution.java
class Solution {
    public long countFairPairs(int[] nums, int lower, int upper) {
        Arrays.sort(nums);
        return countLess(nums, upper + 1) - countLess(nums, lower);
    }
    public long countLess(int[] nums, int val) {
        long res = 0;
        for (int left = 0, right = nums.length - 1; left < right; left++) {
            while (left < right && nums[left] + nums[right] >= val) right--;
            res += right - left;
        }
        return res;
    }
}